tm2/pkg/bft/abci/types/types.go

```diff
@@ -206,6 +206,7 @@ type Error interface {
 
 type Event interface {
 	AssertABCIEvent()
+	SetMsgIdx(int) interface{}
```

Comment:
I'm gonna suggest something radical, but hear me out

We don't actually need this information in the event. 
Why would it matter which event belongs to which message?

How can you guarantee that the node doesn't shuffle the messages before executing them? (it doesn't now, but I see no reason why it might not)

I'm not in favor of this idea because it:
- introduces more complexity on the already bloated event API
- it's just a QoL feature that doesn't have any real functional value, since events are async and we should be concerned with their content, not their origin

Reply:
> We don't actually need this information in the event.
Why would it matter which event belongs to which message?

- Let't say... single block has N txs, and each txs has N2 messages and each message has N3 emitted events.

- If all of N2 messages are same function with just different input params, this is situation where I use `msg_idx` to identify it.

- One of gnoswap(+interface) functionality is removing all of your positions in one shot, by calling `RemovePosition(positionId)` several time(≈several msgs). 

- If that `RemovePosition` emits few event with some underlying infos. At this point, we need to know which events was came from which message (i.e which position emitted which events)

Reply:
> How can you guarantee that the node doesn't shuffle the messages before executing them? (it doesn't now, but I see no reason why it might not)
- I really didn't think about this :(

> introduces more complexity on the already bloated event API
- Agreed

Reply:
> If all of N2 messages are same function with just different input params, this is situation where I use msg_idx to identify it.

Yes, but why? Isn't it enough that you can parse `RemovePosition` events on the client, and use data from the event content to figure out what to do?

Maybe I'm missing something, but tying events to their respective messages doesn't solve any issue -- you should be able to adequately parse them client-side (reactive, async) and execute logic based on their appearance. This is the foundation of [event driven design](https://en.wikipedia.org/wiki/Event-driven_architecture)

@ajnavarro @gfanton maybe I am missing something?